fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/Select.java

```diff
@@ -123,13 +134,18 @@ public void addHavingPredicate(String predicate) {
 
     @Override
     public String toString() {
+        // Just for information purposes...should use the #render call
+        // for database-specific query syntax
         StringBuilder result = new StringBuilder();
         result.append(SELECT);
+        if (this.distinct) {
+            result.append(" DISTINCT");
+        }
         result.append(SPACE).append(this.selectList.toString());
         result.append(SPACE).append(FROM);
         result.append(SPACE).append(this.fromClause.toString());
 
-        if (this.whereClause != null) {
+        if (this.whereClause != null && !this.whereClause.isEmpty()) {
```

Comment:
add isEmpty() checks to following clauses to be consistent

Reply:
ok